sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -641,59 +615,147 @@ func withSuperpowered(superpowered bool) oci.SpecOpts {
 	)
 }
 
-// withRootFileSystemMounts adds container options to mount the root filesystem
-func withRootFilesystemMounts(mountRootFilesystem bool) oci.SpecOpts {
-	// if mountRootFilesystem, return a no-op SpecOpts function
-	if !mountRootFilesystem {
-		return func(_ context.Context, _ oci.Client, _ *containers.Container, s *runtimespec.Spec) error {
-			return nil
-		}
-	}
+// withBootstrap adds container options to grant read-write access to the underlying
+// root filesystem, as well as to manage the devices attached to the
+// host
+func withBootstrap() oci.SpecOpts {
+	return oci.Compose(
+		withPrivilegedMounts(),
+		withRootFsShared(),
+		oci.WithSelinuxLabel("system_u:system_r:control_t:s0"),
+		// Bootstrap containers don't require all "privileges", we only add the
+		// `CAP_SYS_ADMIN` capability. `WithDefaultProfile` will create the proper
+		// seccomp profile based on the container's capabilities.
+		oci.WithAddedCapabilities([]string{"CAP_SYS_ADMIN"}),
+		seccomp.WithDefaultProfile(),
+		oci.WithAllDevicesAllowed,
+	)
+}
 
+// withDefault adds container options for non-privileged containers
+func withDefault() oci.SpecOpts {
 	return oci.Compose(
-		oci.WithMounts([]runtimespec.Mount{
-			{
-				Options:     []string{"rbind", "ro"},
-				Destination: "/.bottlerocket/rootfs",
-				Source:      "/",
-			},
-			{
-				Options:     []string{"rbind", "ro"},
-				Destination: "/lib/modules",
-				Source:      "/lib/modules",
-			},
-			{
-				Options:     []string{"rbind", "rw"},
-				Destination: "/usr/src/kernels",
-				Source:      "/usr/src/kernels",
-			},
-			{
-				Options:     []string{"rbind"},
-				Destination: "/sys/kernel/debug",
-				Source:      "/sys/kernel/debug",
-			},
-		}),
+		oci.WithSelinuxLabel("system_u:system_r:control_t:s0"),
+		// Non-privileged containers only have access to a subset of the devices
+		oci.WithDefaultUnixDevices,
+		// No additional capabilities required for non-privileged containers
+		seccomp.WithDefaultProfile(),
 	)
 }
 
-// withPersistentStorage add persistent storage location that matches the container name
-// (legacy location) and a generically named `current` dir. The `current` dir was added for easier
-// referencing in Dockerfiles and scripts. If a host container is also named `current` this function
-// will only add a single `current` mount to the spec.
-func withPersistentStorage(containerID string, persistentDir string) oci.SpecOpts {
-	var persistentMounts = []runtimespec.Mount{{
-		Options:     []string{"rbind", "rw"},
-		Destination: fmt.Sprintf("/.bottlerocket/%s/%s", persistentDir, containerID),
-		Source:      fmt.Sprintf("/local/%s/%s", persistentDir, containerID),
-	}}
+// withDefaultMounts adds the mount configurations required in all container types
+func withDefaultMounts(containerID string, persistentDir string) oci.SpecOpts {
+	var mounts = []runtimespec.Mount{
+		// Local persistent storage for the container
+		{
+			Options:     []string{"rbind", "rw"},
```

Comment:
Do these mounts need to include `slave` or `rslave` for superpowered/boostrap containers as well?

Reply:
I don't think we should set up those bindings as `slave`s since there aren't any bindings in the containers' persistent storage directory that we want to propagate from the host to the containers' mount namespace.

IMO, we should only propagate mounts down when we want to share them consistently across containers. We shouldn't touch the containers' persistent storage directory since doing so could impact how the users are using that storage in their own containers.